addons/mail/controllers/discuss/avatar_card.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+from odoo.addons.mail.tools.discuss import add_guest_to_context
+
+
+class AvatarCardController(http.Controller):
+    @http.route("/discuss/avatar_card", methods=["POST"], type="jsonrpc", auth="public")
+    @add_guest_to_context
+    def avatar_card(self, user_id, partner_id, fields):
+        if not user_id and partner_id:
+            partner = request.env["res.partner"].browse(partner_id)
+            users = partner.with_context(active_test=False).user_ids
```

Comment:
We should be ignoring non-active users. It's as if they are deleted in terms of features.

Only exception is for OdooBot, which needs to be handled separately.

See compute of `main_user_id` in later versions.